Build failure: dump1090-fa

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

```
nix build .#dump1090-fa
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/319898412

### Relevant log output

```console
dump1090> gcc  -I. -D_POSIX_C_SOURCE=200112L -DMODES_DUMP1090_VERSION=\"10.2\" -DMODES_DUMP1090_VARIANT=\"dump1090-fa\" -D_DEFAULT_SOURCE -DENABLE_CPUFEATURES -Icpu_features/include -DENABLE_RTLSDR -DENABLE_BLADERF -DENABLE_HACKRF -DENABLE_LIMESDR -DENABLE_SOAPYSDR -DSTARCH_MIX_X86 -O3 -g -std=c11 -fno-common -Wall -Wmissing-declarations -Werror -Wformat-signedness -W -I/nix/store/ddwqrs4hxxrwmggjp5v8nvhfb071w0pa-rtl-sdr-blog-1.3.5-dev/include/ -I/nix/store/y6yf9av5r0yz2ppbk0l5ms0f78fpz0wi-libbladeRF-2025.10/include/ -I/nix/store/9lhk4ii7zspip3d81cd8ii30wfcs17i8-hackrf-2024.02.1/include -I/nix/store/9lhk4ii7zspip3d81cd8ii30wfcs17i8-hackrf-2024.02.1/include/libhackrf -I/nix/store/kbin0714jdahqd917p6cpdflwbynibg8-libusb-1.0.29-dev/include/libusb-1.0 -I/nix/store/60v7nwvpph78sfmqb51wxzj5x9wmgksv-limesuite-23.11.0/include -I/nix/store/in6a11lhmw381awgsqqrcfgvfhcqz4kb-soapysdr-0.8.1-unstable-2025-10-05-03/include -c interactive.c -o interactive.o
dump1090> interactive.c: In function 'interactiveShowData':
dump1090> interactive.c:143:23: error: initializer-string for array of 'char' truncates NUL terminator but destination lacks 'nonstring' attribute (5 chars into 4 available) [-Werror=unterminated-string-initialization]
dump1090>   143 |     char spinner[4] = "|/-\\";
dump1090>       |                       ^~~~~~~
dump1090> cc1: all warnings being treated as errors
dump1090> make: *** [Makefile:223: interactive.o] Error 1
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.8, NixOS, 26.05 (Yarara), 26.05.20260203.e6eae2e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.95.0-pre20260203-dev_8b73cbb
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/zitrone/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/zitrone/.local/share/flatpak/exports/etc/xdg/nix/nix.conf:/var/lib/flatpak/exports/etc/xdg/nix/nix.conf:/home/zitrone/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/zitrone/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/zitrone/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/xqbjnjgk7wwqnbd65yvh11c2n88wjpax-lix-2.95.0-pre20260203-dev_8b73cbb/share`
 - nixpkgs: `/nix/store/jjg10vvry6afphbwmx41s0jvsf9rx9wp-source`

### Notify maintainers

@earldouglas @aciceri 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/du/dump1090-fa/package.nix
+++ b/pkgs/by-name/du/dump1090-fa/package.nix
@@ -2,7 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
-  fetchpatch,
+  fetchpatch2,
   pkg-config,
   hackrf,
   libbladeRF,
@@ -23,6 +23,14 @@ stdenv.mkDerivation (finalAttrs: {
     tag = "v${finalAttrs.version}";
     hash = "sha256-kTJ8FMugBRJaxWas/jEj4E5TmVnNpNdhq4r2YFFwgTU=";
   };
+
+  patches = [
+    # Fix compilation with GCC 15: https://github.com/flightaware/dump1090/pull/261
+    (fetchpatch2 {
+      url = "https://github.com/flightaware/dump1090/commit/93be1da123215e8ac15a0deaffedd480e8899f77.patch";
+      hash = "sha256-ehpMfLLEh1pMgvFAPg1JHo8XRlta+GvCIZsSXVPISLc=";
+    })
+  ];
 
   nativeBuildInputs = [ pkg-config ];
 

